packages/assets-controllers/src/TokenBalancesController.ts

```diff
@@ -357,18 +462,34 @@ export class TokenBalancesController extends StaticIntervalPollingController<Tok
       results = await multicallOrFallback(calls, chainId, provider);
     }
 
-    this.update((state) => {
-      // Reset so that when accounts or tokens are removed,
-      // their balances are removed rather than left stale.
-      for (const accountAddress of Object.keys(state.tokenBalances)) {
```

Comment:
Instead of resetting account balances here; we will explicitly handle the case when an account is removed and when a token is removed